parser.go

```diff
@@ -58,6 +58,12 @@ func (p *Parser) ParseWithClaims(tokenString string, claims Claims, keyFunc Keyf
 		return token, err
 	}
 
+	return p.VerifyToken(token, parts, claims, keyFunc)
+}
+
+func (p *Parser) VerifyToken(token *Token, parts []string, claims Claims, keyFunc Keyfunc) (*Token, error) {
```

Comment:
@quantumsheep Can you supply a good for this function? Then we can accept the merge.